icu4c/source/tools/toolutil/writesrc.cpp

```diff
@@ -395,3 +461,29 @@ usrc_writeArrayOfMostlyInvChars(FILE *f,
         fputs(postfix, f);
     }
 }
+
+U_CAPI void U_EXPORT2
+usrc_writeString(FILE *f,
+                 const UChar* ptr, int32_t length,
+                 UTargetSyntax) {
+    UnicodeString ustr(ptr, length);
+    // For now, assume all UTargetSyntax values are valid here.
+    fprintf(f, "\"");
+    for (int32_t i = 0; i < ustr.length(); ) {
+        UChar32 cp = ustr.char32At(i);
+        if (cp == u'"') {
+            fprintf(f, "\\\"");
+        } else if (ICU_Utility::isUnprintable(cp)) {
```

Comment:
You assume that that is true for any cp >= 0x80 (non-ASCII). Might as well just test for that.
Otherwise your code is broken if that function ever returns false for non-ASCII characters.